spec/models/concerns/sample_sheet_spec.rb

```diff
@@ -8,168 +8,225 @@
+        pool = create(:pacbio_pool, :hidden_tagged, library_count: 1)
+        empty_well = create(:pacbio_well, pools: [pool])
+        expect(empty_well.barcode_set).to be_nil
+      end
     end
 
-    it 'returns nothing if the aliquots are tagged with a :hidden tag set (egh. IsoSeq)' do
-      pool = create(:pacbio_pool, :hidden_tagged, library_count: 1)
-      empty_well = create(:pacbio_well, pools: [pool])
-      expect(empty_well.barcode_set).to be_nil
-    end
-  end
+    describe '#sample_is_barcoded' do
+      it 'returns true if all well aliquots are tagged and non isoSeq tags' do
+        expect(well.sample_is_barcoded).to be true
+      end
 
-  describe '#sample_is_barcoded' do
-    it 'returns true if all well aliquots are tagged and non isoSeq tags' do
-      expect(well.sample_is_barcoded).to be true
-    end
+      it 'returns false if there is one aliquots and it has no tag' do
+        pool = create(:pacbio_pool, :untagged, library_count: 1)
+        empty_well = create(:pacbio_well, pools: [pool])
+        expect(empty_well.sample_is_barcoded).to be false
+      end
 
-    it 'returns false if there is one aliquots and it has no tag' do
-      pool = create(:pacbio_pool, :untagged, library_count: 1)
-      empty_well = create(:pacbio_well, pools: [pool])
-      expect(empty_well.sample_is_barcoded).to be false
-    end
+      it 'returns true if there is only one aliquots and it has a non isoSeq tag' do
+        pool = create(:pacbio_pool)
+        empty_well = create(:pacbio_well, pools: [pool])
+        expect(empty_well.sample_is_barcoded).to be true
+      end
 
-    it 'returns true if there is only one aliquots and it has a non isoSeq tag' do
-      pool = create(:pacbio_pool)
-      empty_well = create(:pacbio_well, pools: [pool])
-      expect(empty_well.sample_is_barcoded).to be true
+      it 'returns false if the aliquots are tagged with a :hidden tag set (egh. IsoSeq)' do
+        pool = create(:pacbio_pool, :hidden_tagged, library_count: 1)
+        empty_well = create(:pacbio_well, pools: [pool])
+        expect(empty_well.sample_is_barcoded).to be false
+      end
     end
 
-    it 'returns false if the aliquots are tagged with a :hidden tag set (egh. IsoSeq)' do
-      pool = create(:pacbio_pool, :hidden_tagged, library_count: 1)
-      empty_well = create(:pacbio_well, pools: [pool])
-      expect(empty_well.sample_is_barcoded).to be false
-    end
-  end
+    describe '#bio_sample_name' do
+      context 'when tag set is :default type' do
+        it 'returns nothing if row type is well' do
+          expect(well.bio_sample_name).to be_nil
+        end
 
-  describe '#bio_sample_name' do
-    context 'when tag set is :default type' do
-      it 'returns nothing if row type is well' do
-        expect(well.bio_sample_name).to be_nil
+        it 'returns aliquot sample_name when well has libraries and row type is library' do
+          aliquot = create(:pacbio_library).used_aliquots.first
```

Comment:
I suspect this test is false passing from previous work. By the test title this should create a well with libraries.

Reply:
Not really its probably fine, I wasn't sure the test description matched what it was actually doing but that is a minor problem to have.